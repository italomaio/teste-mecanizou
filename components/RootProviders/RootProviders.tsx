"use client";

import React, { useEffect, useState } from "react";

import dynamic from "next/dynamic";

const NextThemesProvider = dynamic(
  () => import("next-themes").then((mod) => mod.ThemeProvider),
  { ssr: false }
);

type RootProvidersProps = {
  children: React.ReactNode;
};

const RootProviders: React.FC<RootProvidersProps> = ({ children }) => {
  const [mounted, setMounted] = useState<boolean>(false);

  useEffect(() => setMounted(true), []);

  if (!mounted) return <>{children}</>;

  return (
    <NextThemesProvider
      attribute="class"
      defaultTheme="light"
      themes={["light", "dark"]}
      enableSystem={false}
    >
      {children}
    </NextThemesProvider>
  );
};

export default RootProviders;
