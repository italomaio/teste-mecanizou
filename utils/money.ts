export const formatter = new Intl.NumberFormat("pt-BR", {
  style: "currency",
  currency: "BRL",
  minimumFractionDigits: 2,
});

export function calculateDiscountPrice(
  price: number,
  discountPercentage: number
) {
  const discountFactor = discountPercentage / 100;
  const newPrice = price * (1 - discountFactor);

  return {
    oldPrice: formatter.format(price),
    newPrice: formatter.format(newPrice),
  };
}
