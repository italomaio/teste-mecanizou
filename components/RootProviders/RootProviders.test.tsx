import React from "react";
import RootProviders from "./RootProviders";

import { render } from "@/tests/utils";
import { waitFor } from "@testing-library/react";

describe("RootProviders", () => {
  const CHILD_CONTENT = "App Content";

  it("Should render children correctly", async () => {
    const screen = render(
      <RootProviders>
        <div>{CHILD_CONTENT}</div>
      </RootProviders>
    );

    expect(screen.getByText(CHILD_CONTENT)).toBeInTheDocument();

    await waitFor(() => {
      expect(screen.getByTestId("mock-theme-provider")).toBeInTheDocument();
    });
  });

  it("Should render without crashing", () => {
    const { container } = render(
      <RootProviders>
        <div>Another Child</div>
      </RootProviders>
    );

    expect(container).toBeTruthy();
  });

  it("Should match snapshot", () => {
    const { asFragment } = render(
      <RootProviders>
        <div>Snapshot Test</div>
      </RootProviders>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
