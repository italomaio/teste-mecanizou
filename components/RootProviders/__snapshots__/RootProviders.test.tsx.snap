// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`RootProviders Should match snapshot 1`] = `
<DocumentFragment>
  <div
    data-attribute="class"
    data-default-theme="light"
    data-enable-system="false"
    data-testid="mock-theme-provider"
    data-themes="["light","dark"]"
  >
    <div>
      Snapshot Test
    </div>
  </div>
</DocumentFragment>
`;
